common/models/templates/src/common.ts

```diff
@@ -99,4 +99,32 @@ namespace models {
     }
     return suggestion;
   }
+
+  export function defaultApplyCasing(casing: CasingEnum, text: string): string {
+    switch(casing) {
+      case 'lower':
+        return text.toLowerCase();
+      case 'upper':
+        return text.toUpperCase();
+      case 'initial':
+        let headCode = text.charCodeAt(0);
+        // The length of the first code unit, as measured in code points.
+        let headUnitLength = 1;
+  
+        // Is the first character a high surrogate, indicating possible use of UTF-16 
+        // surrogate pairs?  Also, is the string long enough for there to BE a pair?
+        if(text.length > 1 && headCode >= 0xD800 && headCode <= 0xDBFF) {
```

Comment:
How about `isHighSurrogate(headCode)`?

Reply:
... I didn't realize that function was still on the branch; I thought I'd stripped it out.

That said, it'd have been added anyway in the upcoming followup (b/c custom models), so :+1:.